### PR TITLE
[hash-library] Add new port

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?
  Adds new port [hash-library](https://create.stephan-brumme.com/hash-library/).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/hash-library/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
+
+install(
+    EXPORT unofficial-hash-library-targets
+    FILE unofficial-hash-library-targets.cmake
+    NAMESPACE unofficial::
+    DESTINATION share/unofficial-hash-library
+)
```
Comment (Thomas1664) [inline ports/hash-library/CMakeLists.txt:45]: Please add a final newline

File: ports/hash-library/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS_DEBUG
+        -DDISABLE_INSTALL_HEADERS=ON
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-${PORT} CONFIG_PATH share/unofficial-${PORT})
```
Comment (Thomas1664) [inline ports/hash-library/portfile.cmake:24]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME "unofficial-${PORT}" CONFIG_PATH "share/unofficial-${PORT}")

```

**Review by Thomas1664 — COMMENTED** [review]
Minor nits

File: ports/hash-library/CMakeLists.txt (hunk 1 above)
Comment (chausner) [reply ports/hash-library/CMakeLists.txt]: Addressed in c4fcb263b5f30fa16a06bed57aa50c81799cffec.

File: ports/hash-library/portfile.cmake (hunk 2 above)
Comment (chausner) [reply ports/hash-library/portfile.cmake]: Addressed in c4fcb263b5f30fa16a06bed57aa50c81799cffec.

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Comment by BillyONeal** [comment]
@chausner @Thomas1664 This broke CI with:

```
Starting package 895/1846: hash-library:x86-windows
Building package hash-library[core]:x86-windows...
-- Note: hash-library only supports static library linkage. Building static library.
-- Downloading https://github.com/stbrumme/hash-library/archive/hash_library_v8.tar.gz -> stbrumme-hash-library-hash_library_v8.tar.gz...
-- Extracting source D:/downloads/stbrumme-hash-library-hash_library_v8.tar.gz
-- Applying patch 001-fix-macos.patch
-- Using source at D:/buildtrees/hash-library/src/library_v8-6124f7a6ce.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: D:/packages/hash-library_x86-windows/share/hash-library/copyright
-- Performing post-build validation
-- Performing post-build validation done
Uploaded binaries to 1 HTTP remotes.
Installing package hash-library[core]:x86-windows...
The following files are already installed in D:/installed/x86-windows and are in conflict with hash-library:x86-windows

Installed by czmq:x86-windows
    include/sha1.h

Elapsed time for package hash-library:x86-windows: 3.227 s
```

Objections to installing to a subdirectory?

**Comment by BillyONeal** [comment]
https://github.com/microsoft/vcpkg/pull/24433

**Pull request merged**